src/services/post.service.ts

```diff
@@ -12,6 +13,13 @@ class PostService{
     const posted: BasePostDto = await pgDataSource.manager.save(post);
     return posted;
   }
+
+  public async getPostsFromDB(): Promise<BasePostDto[]> {
+    const posts: BasePostDto[] = await PostEntity.find();
+    if (!posts) throw new HttpException(409, 'THERE ARE CURRENTLY NO POSTS');
```

Comment:
I am pretty sure !post will not work if you're checking if the posts is empty. This will probably still return a success but just an empty []. Did you test it using postman? Try and see if an empty post list will throw this exception.

Reply:
gotcha, will change this to check the length of the array instead